Core/include/Acts/Utilities/TrackHelpers.hpp

```diff
@@ -493,6 +497,136 @@ void trimTrack(track_proxy_t track, bool trimHoles, bool trimOutliers,
+      measurementCovariance + filteredCovariance;
+
+  return std::pair(residual, residualCovariance);
+}
+
+template <std::size_t nMeasurementDim,
+          TrackStateProxyConcept track_state_proxy_t>
+std::pair<ActsVector<nMeasurementDim>, ActsSquareMatrix<nMeasurementDim>>
+calculateSmoothedResidual(track_state_proxy_t trackState) {
+  using MeasurementVector = ActsVector<nMeasurementDim>;
+  using MeasurementMatrix = ActsSquareMatrix<nMeasurementDim>;
+
+  auto subspaceHelper =
+      trackState.template fixedBoundSubspaceHelper<nMeasurementDim>();
+
+  MeasurementVector measurement = trackState.calibrated();
+  MeasurementMatrix measurementCovariance = trackState.calibratedCovariance();
+  MeasurementVector smoothed =
+      subspaceHelper.projectVector(trackState.smoothed());
+  MeasurementMatrix smoothedCovariance =
+      subspaceHelper.projectMatrix(trackState.smoothedCovariance());
+
+  MeasurementVector residual = measurement - smoothed;
+  MeasurementMatrix residualCovariance =
+      measurementCovariance + smoothedCovariance;
+
+  return std::pair(residual, residualCovariance);
+}
+
+template <TrackStateProxyConcept track_state_proxy_t>
+double calculatePredictedChi2(track_state_proxy_t trackState) {
+  return visit_measurement(trackState.calibratedSize(), [&](auto N) {
+    constexpr int measdim = decltype(N)::value;
+
+    auto [residual, residualCovariance] =
+        calculatePredictedResidual<measdim>(trackState);
+
+    return residual.transpose() * residualCovariance.inverse() * residual;
+  });
+}
+
+template <TrackStateProxyConcept track_state_proxy_t>
+double calculateFilteredChi2(track_state_proxy_t trackState) {
+  return visit_measurement(trackState.calibratedSize(), [&](auto N) {
+    constexpr int measdim = decltype(N)::value;
+
+    auto [residual, residualCovariance] =
+        calculateFilteredResidual<measdim>(trackState);
+
+    return residual.transpose() * residualCovariance.inverse() * residual;
+  });
+}
+
+template <TrackStateProxyConcept track_state_proxy_t>
+double calculateSmoothedChi2(track_state_proxy_t trackState) {
+  return visit_measurement(trackState.calibratedSize(), [&](auto N) {
+    constexpr int measdim = decltype(N)::value;
+
+    auto [residual, residualCovariance] =
+        calculateSmoothedResidual<measdim>(trackState);
+
+    return residual.transpose() * residualCovariance.inverse() * residual;
+  });
+}
+
+template <TrackStateProxyConcept track_state_proxy_t>
+std::pair<BoundVector, BoundMatrix> calculateUnbiasedParametersCovariance(
+    track_state_proxy_t trackState) {
+  // calculate the unbiased track parameters (i.e. fitted track parameters with
+  // this measurement removed) using Eq.(12a)-Eq.(12c) of NIMA 262, 444 (1987)
+
+  return visit_measurement(trackState.calibratedSize(), [&](auto N) {
+    constexpr int measdim = decltype(N)::value;
+
+    auto H =
+        trackState.projector().template topLeftCorner<measdim, eBoundSize>();
+    auto s = trackState.smoothed();
```

Comment:
Maybe add check that trackState has smoothed and has projector otherwise throw exception to avoid users calling this on wrong state type? Maybe just check that trackState is measurement type?

Reply:
@paulgessinger do you have an opinion on this? should we already throw in `smoothed()` or `calibratedSize`? or should this always be handled downstream?

Reply:
`smoothed` and `calibratedSize` only check as assertions, not explicitly. 

There exists `hasSmoothed`, which could be checked, but that only checks if the allocation is present or not. Equivalently, I believe you can check if `calibratedSize()` is `kInvalid` to check if a measurement is allocated.

Probably the extra check isn't to expensive to just do unconditionally but I never tried this out.